File: src/document-editor/HTMLDocument.ts
import { Document } from './Document';

/**
 * Concrete implementation for HTML documents
 */
export class HTMLDocument extends Document {
  save(filePath: string): string {
    const metadata = Array.from(this.metadata.entries())
      .map(([key, value]) => `    <meta name="${key}" content="${value}">`)
      .join('\n');

    const htmlContent = `<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
${metadata}
    <title>${this.metadata.get('title') || 'Document'}</title>
</head>
<body>
    ${this.content}
</body>
</html>`;

    return `HTML document saved to ${filePath}\n${htmlContent}`;
  }

  display(): string {
    const metadataStr = Array.from(this.metadata.entries())
      .map(([key, value]) => `  ${key}: ${value}`)
      .join('\n');

    return `[HTML Document]
Format: HyperText Markup Language
Metadata:
${metadataStr || '  None'}

Content:
${this.content}
================`;
  }

  getFileExtension(): string {
    return '.html';
  }
}

File: src/main/java/com/sa/carmanagement/factory/HtmlDocument.java
package com.sa.carmanagement.factory;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class HtmlDocument implements Document {
    private String content;

    @Override
    public void save(String content) {
        this.content = content;
        log.info("HtmlDocument: Saving content to HTML format");
    }

    @Override
    public String display() {
        return "<html><body>" + content + "</body></html>";
    }

    @Override
    public String getFormat() {
        return "HTML";
    }
}
